vllm/model_executor/layers/quantization/smoothquant/config.py

```diff
@@ -0,0 +1,306 @@
+
+    def get_min_capability(self) -> int:
+        # TODO: check if this is right.
+        return 80
+
+    @classmethod
+    def get_config_filenames(cls) -> List[str]:
+        """List of filenames to search for in the model directory."""
+        return [
+            "quant_config.json",
+        ]
+
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> "SmoothQuantConfig":
+        layer_format_map: Dict[str, str] = {}
+        for layer_key, format in config.items():
+            if format in FORMAT_REGISTRY:
+                layer_format_map[layer_key] = format
+        return cls(layer_format_map)
+    
+    def get_linear_method(self) -> "SmoothQuantLinearMethod":
+        return SmoothQuantLinearMethod(self)
+
+
+# TODO: why is this needed?
+class Int8GEMM(object):
+    _instance_lock = threading.Lock()
+
+    def __init__(self):
+        if not hasattr(self, "i8cugemm"):
+            self.i8cugemm = ops.I8CUGEMM()
+
+    def __new__(cls, *args, **kwargs):
+        if not hasattr(Int8GEMM, "_instance"):
+            with Int8GEMM._instance_lock:
+                if not hasattr(Int8GEMM, "_instance"):
+                    Int8GEMM._instance = object.__new__(cls)
+        return Int8GEMM._instance
+
+    def get_i8cugemm(self):
+        return self.i8cugemm
+
+
+class SmoothQuantLinearMethod(LinearMethodBase):
+    def __init__(self, sq_config: SmoothQuantConfig) -> None:
+        self.sq_config = sq_config
+        self.sq_type = None
+        self.i8cugemm = Int8GEMM().get_i8cugemm()    
+
+    def maybe_update_loaded_weight_name(self, 
+                                        name: str) -> str:
+        """Convert serialized name k_dequant_scale to dequant_scale.
+
+        This function is called by model_cls.load_weights() during the weight
+        loading process to match on disk state dict to vllm state dict.
+        """
+        if "dequant_scale" in name:
+            suffix = name.split('.')[-1]
+            name.replace(suffix, "dequant_scale")
+        return name
+
+    def scales_shard_splitter(self, 
+                              param: torch.Tensor,
+                              loaded_weight: torch.Tensor, 
+                              shard_id: Union[str, int]) -> Tuple[torch.Tensor, torch.Tensor]:
+        """Index into param for for loading.
+        
+        This function is called by QKVColumnLinear and MergedColumnParallelLinear
+        during weight loading to put the scales from disk in the right spot.
+        """
+        if type(shard_id) == str:
+            qkv_idxs = { "q": 0, "k": 1, "v": 2 }
+            if shard_id not in qkv_idxs:
+                raise ValueError(f"Invalid shard_id {shard_id}")
+            shard_id = qkv_idxs[shard_id]
+        elif type(shard_id) != int:
+            raise ValueError(f"Invalid shard id {shard_id}")
+
+        return param[shard_id], loaded_weight
+
+    def get_layer_format(self, layer_name: str) -> SmoothQuantFormat:
```

Comment:
This should be moved to `SmoothQuantConfig` I think

Reply:
Or at least the core logic should. This api probably still needs to exist, but this logic fits better conceptually with the spirit of the `QuantizationConfig`